metaflow/datastore/task_datastore.py

```diff
@@ -147,6 +147,9 @@ def __init__(
                     )
                     if self.has_metadata(check_meta, add_attempt=False):
                         max_attempt = i
+                    else:
+                        # We can assume that if attempt i does not exist, i + 1 does not either
+                        break
```

Comment:
I don't this assumption is true: It can easily happen that we miss one attempt in the middle